scipy/stats/_continuous_distns.py

```diff
@@ -7806,39 +7813,33 @@ def _rvs(self, df, nc, size=None, random_state=None):
         return random_state.noncentral_chisquare(df, nc, size)
 
     def _logpdf(self, x, df, nc):
-        cond = np.ones_like(x, dtype=bool) & (nc != 0)
```

Comment:
Do you think this was effectively for broadcasting? I'm assuming it passes tests without it, but did you dig any deeper to see if it was needed?

Reply:
I couldn't figure out what the idea behind it was.